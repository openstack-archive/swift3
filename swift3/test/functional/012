#!/bin/bash

# Complete Multipart Upload

. ./common

num=10

S3USER=tester

_gen_complete_xml()
{
    echo '<CompleteMultipartUpload>'

    for i in `seq $num`; do
        echo '<Part>'
        echo "<PartNumber>$i</PartNumber>"
        echo "<ETag>$(_etag $TEST_DIR/segment_50M.dat_$i)</ETag>"
        echo '</Part>'
    done

    echo '</CompleteMultipartUpload>'
}


_s3_put /bucket

upload_id=$(_s3_post /bucket/object?uploads \
    -H "x-amz-acl: public-read-write" \
    -H "x-amz-meta-foo1: 1" \
    -H "x-amz-meta-foo2: 2" \
    -H "x-amz-meta-foo3: 3" | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')

for i in `seq $num`; do
    truncate -s 512K $TEST_DIR/segment_50M.dat_$i
    _s3_put /bucket/object?uploadId=${upload_id}\&partNumber=$i \
        -T $TEST_DIR/segment_50M.dat_$i
done

_gen_complete_xml > $TEST_DIR/complete.xml

_s3_post /bucket/object?uploadId=${upload_id} -T $TEST_DIR/complete.xml -D - | \
    _filter_curl xml

_s3_head /bucket/object | _filter_curl

rm $TEST_DIR/complete.xml

for i in `seq $num`; do
    rm $TEST_DIR/segment_50M.dat_$i
done

