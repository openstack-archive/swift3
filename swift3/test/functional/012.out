QA output created by 012
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X POST -H x-amz-acl: public-read-write -H x-amz-meta-hoge1: 1 -H x-amz-meta-hoge2: 2 -H x-amz-meta-hoge3: 3 http://SWIFT_HOST/bucket/object?uploads... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_00 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=1... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_01 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=2... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_02 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=3... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_03 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=4... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_04 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=5... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_05 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=6... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_06 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=7... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_07 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=8... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_08 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=9... 200
> s3curl --id tester -- -X PUT -T ./segment_50M.dat_09 http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=10... 200
> s3curl --id tester -- -X POST -T complete.xml -D - http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID... 200
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: application/xml
Date: DATE
ETag: "ETAG"
Last-Modified: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Location>http://SWIFT_HOST/bucket/object</Location>
  <Bucket>bucket</Bucket>
  <Key>object</Key>
  <ETag>ETAG</ETag>
</CompleteMultipartUploadResult>
> s3curl --id tester -- -I -X HEAD http://SWIFT_HOST/bucket/object... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: application/octet-stream
Date: DATE
ETag: "ETAG"
Last-Modified: DATE
x-amz-id-2: TXID
x-amz-meta-hoge1: 1
x-amz-meta-hoge2: 2
x-amz-meta-hoge3: 3
x-amz-request-id: TXID
X-Trans-Id: TXID

